services/credentials-service.ts

```diff
@@ -0,0 +1,60 @@
+import * as path from "path";
+import * as crypto from "crypto";
+import { platform, EOL } from "os";
+
+export class CredentialsService implements ICredentialsService {
+	private static ENCRYPTION_ALGORITHM = "aes-256-ctr";
+	private static ENCRYPTION_KEY = "6Gsz97KBp293Q0r0ei4pix98V4PIhm2D";
+	private pathToWindowsCredentialsManager: string;
+
+	constructor(private $childProcess: IChildProcess,
+		private $hostInfo: IHostInfo) {
+		this.pathToWindowsCredentialsManager = path.join(__dirname, "..", "vendor", platform(), "CredentialsManager.exe");
+	}
+
+	public async setCredentials(key: string, credentials: ICredentials): Promise<ICredentials> {
+		if (this.$hostInfo.isWindows) {
+			await this.$childProcess.spawnFromEvent(this.pathToWindowsCredentialsManager, ["set", key, credentials.username, this.encrypt(credentials.password)], "close");
+			return credentials;
+		} else {
+			throw new Error(`Storing credentials is not supported on ${platform()} yet.`);
```

Comment:
I was wondering, why do we have to support both ways to set "hostname", "port" if we don't think of the Authenticated scenario which should be an addition to the standard "proxy" scenario? As I understood I can now have two different proxies - one in config, one in cache. In this case it will use the one from the config + {username,pass} from the cache which is confusing? Have you considered to store only the username/pass in the Credentials Storage and leave the hostname/port as is and manipulate them in the config.json via the command?